src/models/object_detection/ObjectDetection.ts

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from 'react';
+import { Image } from 'react-native';
+import { ETError, getError } from '../../Error';
+import { ObjectDetection } from '../../native/RnExecutorchModules';
+import { Detection } from './types';
+
+interface Props {
+  modelSource: string | number;
+}
+
+interface ObjectDetectionModule {
+  error: string | null;
+  isModelLoading: boolean;
+  isModelGenerating: boolean;
+  forward: (input: string) => Promise<Detection[]>;
+}
+
+export const useObjectDetection = ({
+  modelSource,
+}: Props): ObjectDetectionModule => {
+  const [error, setError] = useState<null | string>(null);
+  const [isModelLoading, setIsModelLoading] = useState(true);
+  const [isModelGenerating, setIsModelGenerating] = useState(false);
+
+  useEffect(() => {
+    const loadModel = async () => {
+      let path = modelSource;
+      if (typeof modelSource === 'number') {
+        path = Image.resolveAssetSource(modelSource).uri;
+      }
+
+      try {
+        setIsModelLoading(true);
```

Comment:
Actually since this in an async function a race condition occurs here, so just move `setIsModelLoading(true)` outside of the `loadModel` callback and call before line 42.